lib/chef/handler/datadog.rb

```diff
@@ -120,6 +121,67 @@ def restore_env_proxies
         ENV['http_proxy'] = @env_http_proxy
         ENV['https_proxy'] = @env_https_proxy
       end
+
+      # create and configure all the Dogapi Clients to be used
+      #
+      # @return [Array] all Dogapi::Client to be used
+      def prepare_the_pack
+        dogs = []
+        endpoints.each do |url, api_key, app_key|
+          dogs.push(Dogapi::Client.new(
+                      api_key,
+                      app_key,
+                      nil,   # host
+                      nil,   # device
+                      true,  # silent
+                      nil,   # timeout
+                      url
+          ))
+        end
+        dogs
+      end
+
+      def convert_to_array(value_or_array)
+        return value_or_array if value_or_array.nil?
+        value_or_array.is_a?(Array) ? value_or_array : [value_or_array]
+      end
+
+      # return all endpoints as a list of triplets [url, api_key, application_key]
+      def endpoints
+        validate_keys(@config[:api_key], @config[:application_key], true)
+        @config[:url] ||= ENV['DATADOG_HOST']
```

Comment:
You're right. FYI, the case I thought of was the following:
`url`, `api_key` and `application_key` set.
`extra_endpoints` with:
```
{
  'endpoint1': {
    'api_key': 'blabla',
    'application_key': 'fdasfa'
  },
  'endpoint2': {
    'api_key': 'blabla2',
    'application_key': 'fdasfa2',
    'url': 'https://app.example.com'
  }
}
```

But actually this doesn't matter for the handler, only for `chef-datadog`.


}